Removes a direct dependency on play

schmitch opened the pull request:
this will actually use slf4j-api, typesafe-config and guice directly
this means that play-mailer is indepdent of play!
in the next steps I will also split out the guice dependency

benmccann commented:
great change!

marcospereira commented on build.sbt:
```diff
@@ -8,21 +8,22 @@ lazy val `play-mailer` = (project in file("."))
 
 libraryDependencies ++= Seq(
-  "com.typesafe.play" %% "play" % PlayVersion % Provided,
+  "com.google.inject" % "guice" % "4.0", // 4.0 to maybe make it work with 2.5 and 2.6
```
This should work for both since for Play 2.6 this version will be evicted. But it would be good to run the tests against both versions of Play here (to ensure that changes made here will work as expected). You can configure Travis to run commands for multiple versions and read `PlayVersion` from some parameter then.

What do you think?

schmitch replied:
hm, yeah I think that might prolly be a good idea. I will definitly look into it.
I guess I can just do it like that:
```
sudo: false
jdk: oraclejdk8
language: scala
matrix:
  include:
  - env: PLAY_VERSION=2.5.15
  - env: PLAY_VERSION=2.6.0-RC2
script: sbt +test
```

and then read the PlayVersion from env? And maybe make a comment that the travis needs to be changed (for tests) if another play version comes in ?

schmitch replied:
Currently for Play 2.5.15 I also need to have bindings for `Config`, however after this PR I want to split the project into play-mailer and play-mailer-guice, and I can maybe add another play-mailer-compat which creates the Config bindings, but is not provided by default (because maybe somebody already have these bindings)

benmccann commented on README.adoc:
```diff
@@ -110,6 +110,9 @@ def sendEmail {
 Since Play 2.4, you can use {runtime-di-uri}[runtime dependency injection] or {compile-time-di-uri}[compile time dependency injection].
 As a result, the `MailerClient` instance can be obtained using runtime dependency injection or compile time dependency injection.
 
+Since Play 2.6, play-mailer is independant from the play version.
```
independant -> independent

play version -> Play version

schmitch commented:
rebased & fix all 
Next Steps (in other PRs):
- split the files and add scalariform
- split guice into it's own module
- change readme from adoc to md and renew it.
